### ISSUE TITLE
Wildcard pattern `_` prevents narrowing based on previous `match` cases

### ISSUE THREAD

**Issue by op200** [astral-sh/ty#4462]
### Summary

```py
def f(a: int | None, b: int | None):
    match a, b:
        case None, None:
            print(1)
        case _, None:
            print(a + 1)
```

```pwsh
> ty check  
error[unsupported-operator]: Unsupported `+` operation
 --> 3.py:8:15
  |
8 |         print(a + 1)
  |               -^^^-
  |               |   |
  |               |   Has type `Literal[1]`
  |               Has type `int | None`

Found 1 diagnostic
```

### Version

ty 0.0.78 (dca9f9873 2026-09-02)

**Comment by carljm** [comment]
Thanks for the report! This requires correlated narrowing of `a` and `b`; the narrowing of `a` is only valid here because both cases require `b` to be `None`. It does seem like pyright does this, but no other type checker does. Would certainly be a nice feature to have.
